main.py

```diff
@@ -1,45 +1,49 @@
-import requests
 import json
 from flask import Flask, request
-app = Flask(__name__)
-
-REQUIRED_LABELS = ['ready for review']
+from models import PullRequest
+from config import REQUIRED_LABELS_ALL, REQUIRED_LABELS_ANY, BANNED_LABELS
 
-# config = {
-#     "all": ["label1", "label2"],
-#     "not": ["label3"],
-#     "any": ["label4", "label5"]
-# }
+app = Flask(__name__)
 
 
 @app.route('/', methods=["POST"])
 def main():
-    pull_request = PullRequest(request.get_json())
-    return create_status_json(any(l['name'] in REQUIRED_LABELS for l in pull_request.labels))
+    event_json = request.get_json()
+    if event_warrants_label_check(event_json):
+        pull_request = PullRequest(event_json)
+        status_code = pull_request.post_status(
+            create_status_json(pull_request, REQUIRED_LABELS_ANY, REQUIRED_LABELS_ALL, BANNED_LABELS))
+        return str(status_code)
+    else:
+        return 'No label check needed'
 
 
-def create_status_json(has_required_labels):
-    if has_required_labels:
-        description = "Your PR has the necessary labels"
+def event_warrants_label_check(pr_event_json):
+    return pr_event_json['action'] in ['opened', 'reopened', 'labeled', 'unlabeled']
+
+
+def create_status_json(pull_request, required_any, required_all, banned):
```

Comment:
would this be better suited as an instance method of the `PullRequest` class? It is dependent on the `pull_request` and you pass the result into the `post_status` method of the `PullRequest` anyway...